wp-content/themes/core/components/header/subheader_archive/Subheader_Archive_Controller.php

```diff
@@ -0,0 +1,172 @@
+	private string $title;
+	private string $description;
+	private int $hero_image_id;
+
+	public function __construct( array $args = [] ) {
+		$args = $this->parse_args( $args );
+
+		$this->classes           = (array) $args[ self::CLASSES ];
+		$this->attrs             = (array) $args[ self::ATTRS ];
+		$this->container_classes = (array) $args[ self::CONTAINER_CLASSES ];
+		$this->media_classes     = (array) $args[ self::MEDIA_CLASSES ];
+		$this->content_classes   = (array) $args[ self::CONTENT_CLASSES ];
+		$this->title             = '';
+		$this->description       = '';
+		$this->map_acf_fields();
+	}
+
+	protected function defaults(): array {
+		return [
+			self::CLASSES           => [ 'c-subheader' ],
+			self::CONTAINER_CLASSES => [ 'l-container' ], 
+			self::ATTRS             => [],
+			self::MEDIA_CLASSES     => [],
+			self::CONTENT_CLASSES   => [],
+		];
+	}
+
+	protected function required(): array {
+		return [
+			self::CONTAINER_CLASSES => [ 'l-container' ], 
+			self::MEDIA_CLASSES     => [ 'c-subheader__media' ],
+			self::CONTENT_CLASSES   => [ 'c-subheader__content' ],
+			self::CLASSES           => [ 'c-subheader', 'c-subheader-archive' ],
+		];
+	}
+
+	public function get_classes(): string {
+		if ( has_post_thumbnail() ) {
+			$this->classes[] = 'c-subheader--has-image';
+		}
+
+		return Markup_Utils::class_attribute( $this->classes );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_content_classes(): string {
+		return Markup_Utils::class_attribute( $this->content_classes );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_media_classes(): string {
+		return Markup_Utils::class_attribute( $this->media_classes );
+	}
+
+
+	/**
+	 * @return string
+	 */
+	public function get_container_classes(): string {
+		return Markup_Utils::class_attribute( $this->container_classes );
+	}
+
+	public function get_attrs(): string {
+		return Markup_Utils::concat_attrs( $this->attrs );
+	}
+
+	
+	/**
+	 * @return array
+	 */
+	public function get_image_args(): array {
+		if (  empty( $this->hero_image_id ) ) {
+			return [];
+		}
+	
+		return [
+			Image_Controller::IMG_ID       => (int) $this->hero_image_id,
+			Image_Controller::AUTO_SHIM    => false,
+			Image_Controller::USE_LAZYLOAD => true,
+			Image_Controller::CLASSES      => [ 'c-image--overlay', 'c-image--object-fit' ],
+			Image_Controller::IMG_CLASSES  => [ 'c-subheader__media__image' ],
+			Image_Controller::SRC_SIZE     => Image_Sizes::SIXTEEN_NINE,
+			Image_Controller::SRCSET_SIZES => [
+				Image_Sizes::SIXTEEN_NINE,
+				Image_Sizes::SIXTEEN_NINE_SMALL,
+			],
+		];
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_description_args(): array {
+		return [
+			Text_Controller::TAG     => 'p',
+			Text_Controller::CLASSES => [ 'c-subheader__description' ],
+			Text_Controller::CONTENT => $this->description,
+		];
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_title_args(): array {
+		return [
+			Text_Controller::TAG     => 'h1',
```

Comment:
It should always be an `<h1>` tag, so no need to give the option, but if I omit it, by default Text_Controller uses a `<p>` tag. Then if I add the `<h1>` directly on the markup, it will display it like `<h1><p>title</p><h1>` (not correct). 

So I don't really know the best option here.

Reply:
Cool, was just wondering. If you're confident it should be an h1, then let's do that.